core/domain/html_validation_service.py

```diff
@@ -769,35 +787,34 @@ def add_dimensions_to_noninteractive_image_tag(exp_id, html_string):
     """
     soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
     for image in soup.findAll(name='oppia-noninteractive-image'):
-        filename = unescape_html(image['filepath-with-value'])
-        filename = filename[1:-1]
+        filename = json.loads(unescape_html(image['filepath-with-value']))
         image['filepath-with-value'] = escape_html(json.dumps(
             get_filename_with_dimensions(filename, exp_id)))
-    return unicode(soup)
+    return unicode(soup).replace('<br/>', '<br>')
 
 
-def get_filename_with_dimensions(filename, exp_id):
-    """Gets the dimensions of the image file.
+def get_filename_with_dimensions(old_filename, exp_id):
+    """Gets the filename with dimensions of the image file in it.
 
     Args:
-        filename: str. Name of the file whose dimensions need to be
+        old_filename: str. Name of the file whose dimensions need to be
             calculated.
         exp_id: str. Exploration id.
 
     Returns:
-        dict. A dict representing Fileinfo object of the image.
+        str. The new filename of the image file.
     """
     file_system_class = (
         fs_domain.ExplorationFileSystem if feconf.DEV_MODE
         else fs_domain.GcsFileSystem)
     fs = fs_domain.AbstractFileSystem(file_system_class(exp_id))
     filepath = (
-        filename if feconf.DEV_MODE
-        else ('image/%s' % filename))
+        old_filename if feconf.DEV_MODE
+        else ('image/%s' % old_filename))
     content = fs.get(filepath)
     height, width = gae_image_services.get_image_dimensions(content)
-    filename_wo_filetype = filename[:filename.rfind('.')]
-    filetype = filename[filename.rfind('.') + 1:]
-    renamed_filename = '%s_height_%s_width_%s.%s' % (
+    filename_wo_filetype = old_filename[:old_filename.rfind('.')]
```

Comment:
You should reuse regenerate_image_filename_using_dimensions() here.

Reply:
Done